### PR TITLE
[Dubbo-3370]Support merger annotation

### PR DESCRIPTION
**Pull request by xxlv**
## What is the purpose of the change

Add merger for Reference annotation

This PR is related to this Issue ( #3370 ) 

## Brief changelog

org.apache.dubbo.config.AbstractReferenceConfig
org.apache.dubbo.config.annotation.Reference
org.apache.dubbo.config.spring.beans.factory.annotation.AnnotationPropertyValuesAdapterTest

## Verifying this change

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/incubator-dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [x] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/incubator-dubbo/tree/master/dubbo-test).
- [ ] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/incubator-dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by ralf0131** [comment]
Hi, sorry to keeping your pull request too long. Would you please resolve the conflict first?

**Comment by hsqlu** [comment]
why this pr hasn't been merged? Are there still some conflicts?

File: dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/annotation/Reference.java [hunk 1]
```diff
@@ -1,265 +1,270 @@
...
-     */
-    String protocol() default "";
-
-    /**
-     * methods support
-     * @return
-     */
-    Method[] methods() default {};
-}
+/*
```
Comment (AlbumenJ) [inline dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/annotation/Reference.java:1]: pls recover end-line format of this file

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

**Comment by AlbumenJ** [comment]
@xxlv hi, thanks for your contribution

pls resolve conflicts with the latest master branch

**Comment by AlbumenJ** [comment]
Close for long time no response.
Please feel free to reopen if you have any question.

**Pull request closed without merging**